### PR TITLE
Fixing a bug where empty variables in .env files were read as multi-line values (#3934)

### PR DESCRIPTION
**Pull request by joehan**
### Description
Fixes an issue where the .env parser would mishandled empty values. The following .env file:
```
FOO=
BAR=something
```
would be parsed into
```
{ FOO: "BAR=something" }
```
Fixes https://github.com/firebase/firebase-tools/issues/3934

### Scenarios Tested
Added a test case to cover this

### CODE REVIEW

File: src/test/functions/env.spec.ts [hunk 1]
```diff
@@ -135,26 +135,33 @@ BAR=bar
...
+      //       {
+      //         description: "should ignore empty lines",
+      //         input: `
+      // FOO=foo
+
+      // BAR=bar
+
+      // `,
+      //         want: { FOO: "foo", BAR: "bar" },
+      //       },
```
Comment (taeold) [inline src/test/functions/env.spec.ts:164]: Do we want to uncomment this?

File: src/functions/env.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ const LINE_RE = new RegExp(
   "(" +                      //   begin optional value
   "\\s*'(?:\\\\'|[^'])*'|" + //     single quoted or
   '\\s*"(?:\\\\"|[^"])*"|' + //     double quoted or
-  "[^#\\r\\n]+" +            //     unquoted
+  "[^#=\\r\\n]*" +           //     unquoted
```
Comment (taeold) [inline src/functions/env.ts:52]: q: Did we need `=`?

IIUC, we are making something like

```
FOO=bar=a
```

from being a valid entry (`{FOO: "bar=a"}`) to an invalid one.

**Review by taeold — CHANGES_REQUESTED** [review]

File: src/test/functions/env.spec.ts (hunk 1 above)
Comment (joehan) [reply src/test/functions/env.spec.ts:164]: 🤦  Forgot to save the file before commiting

File: src/functions/env.ts (hunk 2 above)
Comment (joehan) [reply src/functions/env.ts:52]: Hmmm, its a bit of a weird case - i guess what we really want to do is allow =, but not after a line break?
So something like ```
[[^#\\r\\n][^#=\\r\\n]]
```?
There may be other regex tricks that could solve this as well - maybe replacing the `\\s` that. matches whitespace after the equals with something that doesn't match newlines?

Comment (joehan) [reply src/functions/env.ts:52]: Ahhhh, i just read your original code comments more closely, and realized we explicitly don't support multiline unquoted values. I think i know how to fix this, will play around a bit and make sure all the tests work.

**Review by taeold — APPROVED** [review]

**Pull request merged**